value/src/main/java/com/google/auto/value/processor/TemplateVars.java

```diff
@@ -111,6 +132,17 @@ static Template parsedTemplateForResource(String resourceName) {
     }
   }
 
+  private static String getJARFromURL(Class<?> clazz) {
+    URL url = clazz.getResource('/' + clazz.getName().replace('.', '/') + ".class");
```

Comment:
FWIW, there's no guarantee that you can load classes as resource this way, though in practice you can with URLClassLoader.